packages/block-library/src/media-text/edit.js

```diff
@@ -40,6 +55,202 @@ const TEMPLATE = [
 const WIDTH_CONSTRAINT_PERCENTAGE = 15;
 const applyWidthConstraints = ( width ) => Math.max( WIDTH_CONSTRAINT_PERCENTAGE, Math.min( width, 100 - WIDTH_CONSTRAINT_PERCENTAGE ) );
 
+export const LINK_DESTINATION_NONE = 'none';
+export const LINK_DESTINATION_CUSTOM = 'custom';
+export const LINK_DESTINATION_MEDIA = 'media';
+export const LINK_DESTINATION_ATTACHMENT = 'attachment';
+export const NEW_TAB_REL = [ 'noreferrer', 'noopener' ];
+
+const icon = <SVG viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><Path d="M0,0h24v24H0V0z" fill="none" /><Path d="m19 5v14h-14v-14h14m0-2h-14c-1.1 0-2 0.9-2 2v14c0 1.1 0.9 2 2 2h14c1.1 0 2-0.9 2-2v-14c0-1.1-0.9-2-2-2z" /><Path d="m14.14 11.86l-3 3.87-2.14-2.59-3 3.86h12l-3.86-5.14z" /></SVG>;
+const stopPropagation = ( event ) => {
+	event.stopPropagation();
+};
+
+const stopPropagationRelevantKeys = ( event ) => {
+	if ( [ LEFT, DOWN, RIGHT, UP, BACKSPACE, ENTER ].indexOf( event.keyCode ) > -1 ) {
+		// Stop the key event from propagating up to ObserveTyping.startTypingInTextField.
+		event.stopPropagation();
+	}
+};
+
+const removeNewTabRel = ( currentRel ) => {
+	let newRel = currentRel;
+
+	if ( currentRel !== undefined && ! isEmpty( newRel ) ) {
+		if ( ! isEmpty( newRel ) ) {
+			each( NEW_TAB_REL, function( relVal ) {
+				const regExp = new RegExp( '\\b' + relVal + '\\b', 'gi' );
+				newRel = newRel.replace( regExp, '' );
+			} );
+
+			// Only trim if NEW_TAB_REL values was replaced.
+			if ( newRel !== currentRel ) {
+				newRel = newRel.trim();
+			}
+
+			if ( isEmpty( newRel ) ) {
+				newRel = undefined;
+			}
+		}
+	}
+
+	return newRel;
+};
+
+const ImageURLInputUI = ( {
```

Comment:
This component is the same as the one used in the image block. Given that now we have multiple use cases, could we create a generic experimental component part of block-editor that is used by both blocks?

Reply:
Yes we can!

Reply:
Done @jorgefilipecosta